### PR TITLE
Sandbox Environment

### PR DESCRIPTION
**Pull request by asteel-gsa**
This is a fairly extensive PR that I would like to have 1-2 more folks test before merging. This has been tested, built, destroyed, and rebuilt by Matt as well as myself.

### Description
Fundamentally, nothing changes except the following:
- Adds `helper_scripts` directory to `sandbox` to allow for easy `./` commands to plan, apply and destroy the environment
- Creates a clone of `https-proxy` called `sandbox-proxy` as the core change to this is what module manages the routes
  - Routes now located in `shared/sandbox/routes.tf` rather than where they were (clamav and https-proxy)
-  Detailed README on how to use sandbox
- Adds `"SANDBOX"` to `settings.py`

### CODE REVIEW

File: terraform/shared/modules/https-proxy/prepare-proxy.sh [hunk 1]
```diff
@@ -13,11 +13,12 @@ tmpdir=$(mktemp -d 2>/dev/null || mktemp -d -t 'mytmpdir')
 cd "$tmpdir"
 
 # Grab a copy of the zip file for the specified ref
-curl -s -L https://github.com/GSA-TTS/cg-egress-proxy/archive/${GITREF}.zip --output local.zip 
+curl -s -L https://github.com/GSA-TTS/cg-egress-proxy/archive/${GITREF}.zip --output local.zip
 
 # Zip up just the proxy/ subdirectory for pushing
 unzip -q -u local.zip \*/proxy/\*
-zip -q -j -r ${popdir}/proxy.zip cg-egress-proxy-*/proxy
+cd "${tmpdir}/cg-egress-proxy-main/proxy/" && zip -r -o -X "${popdir}/proxy.zip" ./ > /dev/null
```
Comment (asteel-gsa) [inline terraform/shared/modules/https-proxy/prepare-proxy.sh:20]: Can't remember why I made this change. I think it makes sense, but will confirm locally and in preview

Comment (asteel-gsa) [reply terraform/shared/modules/https-proxy/prepare-proxy.sh:20]: NVM, can remove this change. Think I forgot about it after I made the `sandbox-proxy` module clone

File: terraform/shared/modules/stream-proxy/prepare-proxy.sh [hunk 2]
```diff
@@ -1,11 +1,9 @@
 #!/bin/sh
 
 # Exit if any step fails
-set -e
```
Comment (asteel-gsa) [inline terraform/shared/modules/stream-proxy/prepare-proxy.sh:4]: Need to undo this

File: terraform/shared/modules/stream-proxy/stream-proxy.tf [hunk 3]
```diff
@@ -1,97 +1,97 @@
...
+###
+
+resource "cloudfoundry_user_provided_service" "credentials" {
+  name  = "${var.name}-creds"
+  space = data.cloudfoundry_space.client_space.id
+  credentials = {
+    "domain" = cloudfoundry_route.egress_route.endpoint
+    "port"   = 8080
+  }
+}
```
Comment (asteel-gsa) [inline terraform/shared/modules/stream-proxy/stream-proxy.tf:97]: CRLF -> LF Change

File: backend/Procfile [hunk 4]
```diff
@@ -1,2 +1,2 @@
-# used by cloud.gov
-web: newrelic-admin run-program gunicorn config.wsgi -c gunicorn.conf.py
+# used by cloud.gov
+web: newrelic-admin run-program gunicorn config.wsgi -c gunicorn.conf.py
```
Comment (asteel-gsa) [inline backend/Procfile:2]: CRLF -> LF Change (I think)

File: terraform/shared/modules/stream-proxy/prepare-proxy.sh (hunk 2 above)
Comment (asteel-gsa) [reply terraform/shared/modules/stream-proxy/prepare-proxy.sh]: Fixed

**Comment by sambodeme** [comment]
After an initial walkthrough session with Alex, I ran destroy.sh, followed by plan.sh and apply.sh. The apply command initially failed with an error that appeared to be a random blip, so I reran plan.sh and apply.sh, and this time it worked. Wanted to do a complete rerun again from the start.

**Comment by rnovak338** [comment]
Had a successful runthrough with Alex last week myself, and yesterday I was able to go through README on my own time and get the helper scripts running successfully.

**Review by sambodeme — APPROVED** [review]

**Pull request merged**